hazelcast-sql/src/main/java/com/hazelcast/sql/impl/schema/type/Type.java

```diff
@@ -151,54 +81,15 @@ public String unparse() {
     @Override
     public void writeData(final ObjectDataOutput out) throws IOException {
         out.writeString(name);
-        out.writeInt(kind.ordinal());
-        switch (kind) {
-            case JAVA:
-                out.writeString(javaClassName);
-                break;
-            case PORTABLE:
-                out.writeInt(portableFactoryId);
-                out.writeInt(portableClassId);
-                out.writeInt(portableVersion);
-                break;
-            case COMPACT:
-                out.writeString(compactTypeName);
-                break;
-            default:
-                throw new UnsupportedOperationException("Unsupported Type Kind: " + kind);
-        }
-
-        out.writeInt(fields.size());
-        for (final TypeField field : fields) {
-            out.writeObject(field);
-        }
+        out.writeObject(fields);
```

Comment:
breaking change should be mentioned in the PR description. Doesn't it break some upgrade compatilibility tests? Do we have upgrade compatibility tests for UDTs at all?

Reply:
I mentioned the breaking change in the PR description. There is no upgrade compatibility tests for UDTs, but we will need to add them after making UDTs production ready. That's why I am unwilling to make this change since we will not easily be able to introduce such breaking changes afterwards.

Reply:
without "UDTs production ready" we would have also to make nested fields support experimental/beta